internal/providers/service.go

```diff
@@ -57,13 +57,52 @@ type ProviderService interface {
 // from the state
 var ErrInvalidTokenIdentity = errors.New("invalid token identity")
 
+// GitHubClientService is an interface for GitHub operations inside this module
+// It is used to mock GitHub operations in tests, but in order to generate
+// mocks, the interface must be exported
+type GitHubClientService interface {
+	GetInstallation(ctx context.Context, id int64, jwt string) (*github.Installation, *github.Response, error)
+	GetUserIdFromToken(ctx context.Context, token *oauth2.Token) (*int64, error)
+	ListUserInstallations(ctx context.Context, token *oauth2.Token) ([]*github.Installation, error)
+}
+
+type provSvcGhOpsImpl struct{}
+
+func (provSvcGhOpsImpl) GetInstallation(
+	ctx context.Context,
+	installationID int64,
+	jwt string,
+) (*github.Installation, *github.Response, error) {
+	ghClient := github.NewClient(nil).WithAuthToken(jwt)
+	return ghClient.Apps.GetInstallation(ctx, installationID)
+}
+
+func (provSvcGhOpsImpl) GetUserIdFromToken(ctx context.Context, token *oauth2.Token) (*int64, error) {
+	ghClient := github.NewClient(nil).WithAuthToken(token.AccessToken)
+
+	user, _, err := ghClient.Users.Get(ctx, "")
+	if err != nil {
+		return nil, err
+	}
+
+	return user.ID, nil
+}
+
+func (provSvcGhOpsImpl) ListUserInstallations(ctx context.Context, token *oauth2.Token) ([]*github.Installation, error) {
+	ghClient := github.NewClient(nil).WithAuthToken(token.AccessToken)
+
+	installations, _, err := ghClient.Apps.ListUserInstallations(ctx, nil)
+	return installations, err
+}
+
 type providerService struct {
 	store           db.Store
 	cryptoEngine    crypto.Engine
 	mt              metrics.Metrics
 	provMt          provtelemetry.ProviderMetrics
 	config          *server.ProviderConfig
 	restClientCache ratecache.RestClientCache
+	provSvcGhOps    GitHubClientService
```

Comment:
Do you want to rename the field here as well?

Reply:
good find, on it

Reply:
@eleftherias done!